src/lib/y2network/hwinfo.rb

```diff
@@ -184,6 +237,17 @@ def drivers
       modules.map { |m| Driver.new(*m) }
     end
 
+    # Determines whether the hardware is available (plugged)
+    #
+    # If the hardware layer was able to get its type, it consider the hardware to be connected. Bear
+    # in mind that it is not possible to just rely in #exists? because it could include some info
+    # from udev rules.
+    #
+    # @return [Boolean]
+    def present?
+      !!type
```

Comment:
is type really so deterministic? I though that type could be part of udev rule, not? What about dev_name?

Reply:
I would say that the `type` is always there (but perhaps there is a better candidate) and, from udev, we are only able to infer the bus ID and the MAC. On the other hand, we want to avoid the dev_name because it might be missing (s390) and that's the main problem we try to address.